pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -142,34 +140,30 @@ func (p *ldapProvider) getPrincipalsFromSearchResult(result *ldapv3.SearchResult
 	userPrincipal = *user
 	userDN := result.Entries[0].DN
 
-	if len(userMemberAttribute) > 0 {
-		for i := 0; i < len(userMemberAttribute); i += 50 {
-			batchGroupDN := userMemberAttribute[i:min(i+50, len(userMemberAttribute))]
-			filter := fmt.Sprintf("(%v=%v)", ObjectClass, config.GroupObjectClass)
-			query := "(|"
-			for _, gdn := range batchGroupDN {
-				query += fmt.Sprintf("(%v=%v)", config.GroupDNAttribute, ldapv3.EscapeFilter(gdn))
-			}
-			query += ")"
-			query = fmt.Sprintf("(&%v%v)", filter, query)
-			// Pulling user's groups
-			logrus.Debugf("Ldap: Query for pulling user's groups: %v", query)
-			userMemberGroupPrincipals, err := p.searchLdap(query, groupScope, config, lConn)
-			groupPrincipals = append(groupPrincipals, userMemberGroupPrincipals...)
-			if err != nil {
-				return userPrincipal, groupPrincipals, err
-			}
+	for i := 0; i < len(userMemberAttribute); i += 50 {
+		batchGroupDN := userMemberAttribute[i:min(i+50, len(userMemberAttribute))]
+		filter := fmt.Sprintf("(%v=%v)", ObjectClass, config.GroupObjectClass)
+		query := "(|"
+		for _, gdn := range batchGroupDN {
+			query += fmt.Sprintf("(%v=%v)", config.GroupDNAttribute, ldapv3.EscapeFilter(gdn))
+		}
+		query += ")"
+		query = fmt.Sprintf("(&%v%v)", filter, query)
+		// Pulling user's groups
+		logrus.Debugf("Ldap: Query for pulling user's groups: %v", query)
+		userMemberGroupPrincipals, err := p.searchLdap(query, groupScope, config, lConn)
+		groupPrincipals = append(groupPrincipals, userMemberGroupPrincipals...)
+		if err != nil {
+			return userPrincipal, groupPrincipals, err
```

Comment:
We should probably return zero values instead 🤔 
```suggestion
			return v3.Principal{}, nil, err
```